### PR TITLE
refactor of beta function plus unit tests to cover the changes

### PR DESCRIPTION
**Pull request by zangelus**
As discussed, the refactored beta function that makes use of the added common components to the project's library and the corresponding unit tests

### CODE REVIEW

File: SuperCalculator/src/UnitTests/BetaUnitTests.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    @Test
+    public void TestBetaValuesWithinTheDomainMapToCorrespondingValuesInTheCoDomain() throws InvalidDomainException {
+        
+        Beta beta = new Beta(1, 2);
+        assertEquals(0.5d, beta.calculate(), java.lang.Math.pow(1, -15));
+
+        beta = new Beta(15, 19);
+        assertEquals(6.4278e-11, beta.calculate(), java.lang.Math.pow(1, -15));
+        
+        beta = new Beta(2, 2);
```
Comment (Diriks) [inline SuperCalculator/src/UnitTests/BetaUnitTests.java:28]: Are you testing symmetry for the Beta function?

File: SuperCalculator/src/supercalculator/Beta.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        }
+        catch (ArithmeticException ae) {
+            
+            System.out.println("ArithmeticException occured!");
+            result = Double.POSITIVE_INFINITY;
+        }
+
+        return result;
+    }
+}
```
Comment (Diriks) [inline SuperCalculator/src/supercalculator/Beta.java:103]: Beta.java, looks good. Proper. programming standards are adhered to

File: SuperCalculator/src/supercalculator/InvalidDomainException.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ *
+ * @author za
+ */
+public class InvalidDomainException extends Exception{
+    
+    public InvalidDomainException(String message){
+        super(message);
+    }
+}
```
Comment (Diriks) [inline SuperCalculator/src/supercalculator/InvalidDomainException.java:17]: This looks good as well

File: SuperCalculator/src/UnitTests/BetaUnitTests.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    @Test
+    public void TestBetaValuesWithinTheDomainMapToCorrespondingValuesInTheCoDomain() throws InvalidDomainException {
+        
+        Beta beta = new Beta(1, 2);
+        assertEquals(0.5d, beta.calculate(), java.lang.Math.pow(1, -15));
+
+        beta = new Beta(15, 19);
+        assertEquals(6.4278e-11, beta.calculate(), java.lang.Math.pow(1, -15));
+
```
Comment (Diriks) [inline SuperCalculator/src/UnitTests/BetaUnitTests.java:27]: test cases look good

**Review by Diriks — COMMENTED** [review]
All files have been reviewed and were found satisfactory.

File: SuperCalculator/src/UnitTests/BetaUnitTests.java (hunk 1 above)
Comment (zangelus) [reply SuperCalculator/src/UnitTests/BetaUnitTests.java:28]: I cover the symmetry of the function beta in a test below.

**Pull request merged**